### DISCUSSION TITLE
Can Sveltekit be used to build a full backend?

### DISCUSSION THREAD

**Discussion by shindradavid** [sveltejs/kit#11134] [Q&A]
I hope you're all doing well. I've been exploring the possibility of using SvelteKit as a full backend for a web application, particularly when paired with Drizzle ORM and PostgreSQL. I'm intrigued by the idea of having a unified framework for both the frontend and backend, and SvelteKit seems like a promising candidate.

The application I am building is a payment gateway with a dashboard and an api that merchants can use to collect payments from their customers. API authentication will be handled with API keys. My questions are, would it be better to build a separate backend with specialized framework like Express or is Sveltekit capable? What differentiates Sveltekit from backend frameworks like Django?

Thank you all for your time and expertise!

**Comment by david-plugge** [comment]
There are huge benefits for using sveltekit as fullstack framework.

It's made for that use case. 
You get type safety for data you want to display by using the load functions and for mutations by using form actions.
You can co-locate your logic and ui.
You can easily share utility functions since it's one single codebase.
You use the same language for front and backend.
You need only one documentation.

I've build several projects with sveltekit as a fullstack framework with tools like Lucia Auth, drizzle or prisma, Inlang and many more.
In my opinion it's the best way to build stuff IF you don't plan to expose your api. If you want to build an api that will be consumed by external clients I'd rather go with something different (I like elysia). Something that has builtin swagger support,...
